In [4]:
file_path_1 = "./results/MG_323757.bin1/MG_323757.bin1.json"
file_path_2 = "./results/MG_323757.bin2/MG_323757.bin2.json"
file_path_3 = "./results/MG_323757.bin3/MG_323757.bin3.json"

In [5]:
def above_two(data_records):
    index=[]
    for i in range(len(data_records)):
        modules = data_records[i]["modules"].keys()
        if len(modules) > 2:
            # print(i, len(modules))
            index.append(i)
        else:
            continue
    return index

In [6]:
import json
with open(file_path_1, 'r') as files:
    data = json.load(files)
    
    file_name = data["input_file"]
    
print(data.keys())

dict_keys(['version', 'input_file', 'records', 'timings', 'taxon', 'schema'])


In [7]:
data["input_file"]

'MG_323757.bin1.fa'

In [8]:
data["timings"]

{'NODE_83_length_8923_cov_7.096640': {'antismash.detection.full_hmmer': 3.5509538650512695,
  'antismash.detection.hmm_detection': 0.23520898818969727,
  'antismash.detection.cluster_hmmer': 2.271585464477539,
  'antismash.detection.genefunctions': 0.43398451805114746,
  'antismash.detection.nrps_pks_domains': 0.06709933280944824,
  'antismash.detection.tigrfam': 1.3289079666137695,
  'antismash.modules.active_site_finder': 0.0024826526641845703,
  'antismash.modules.cluster_compare': 2.4095418453216553,
  'antismash.modules.clusterblast': 30.847227096557617,
  'antismash.modules.lanthipeptides': 0.18405699729919434,
  'antismash.modules.lassopeptides': 0.19834327697753906,
  'antismash.modules.nrps_pks': 6.175041198730469e-05,
  'antismash.modules.pfam2go': 0.01813030242919922,
  'antismash.modules.rrefinder': 0.015251874923706055,
  'antismash.modules.sactipeptides': 0.19112730026245117,
  'antismash.modules.smcog_trees': 0.9665520191192627,
  'antismash.modules.t2pks': 2.62260437011

In [9]:
data["records"][81]["modules"]["antismash.detection.nrps_pks_domains"]

{'cds_results': {},
 'schema_version': 4,
 'record_id': 'NODE_110_length_7016_cov_6.405832'}

In [10]:
data["records"][81]["modules"]["antismash.modules.lanthipeptides"]

{'record_id': 'NODE_110_length_7016_cov_6.405832',
 'schema_version': 5,
 'motifs': {},
 'new_cds_features': [],
 'protoclusters': {},
 'comparippson': {'db_results': [{'database': {'name': 'MIBiG',
     'version': '3.1',
     'url': 'https://mibig.secondarymetabolites.org/go/@accession@',
     'id_format': '@accession@ (@locus@)',
     'description_format': '@type@: @compounds@',
     'fields': ['accession', 'compounds', 'type', 'locus'],
     'dir_name': 'mibig'},
    'hits': {},
    'aliases': {}},
   {'database': {'name': 'antiSMASH-DB',
     'version': '4.0',
     'url': 'https://antismash-db.secondarymetabolites.org/area.html?record=@accession@&start=@start@&end=@end@',
     'id_format': '@accession@',
     'description_format': '@type@: @locus@',
     'fields': ['accession', 'type', 'locus', 'start', 'end'],
     'dir_name': 'asdb'},
    'hits': {},
    'aliases': {}}],
  'aliases': {}}}

In [11]:
data["records"][81]["modules"]["antismash.modules.lassopeptides"]

{'record_id': 'NODE_110_length_7016_cov_6.405832',
 'schema_version': 3,
 'motifs': {},
 'new_cds_features': [],
 'protoclusters': {},
 'comparippson_results': {'db_results': [{'database': {'name': 'MIBiG',
     'version': '3.1',
     'url': 'https://mibig.secondarymetabolites.org/go/@accession@',
     'id_format': '@accession@ (@locus@)',
     'description_format': '@type@: @compounds@',
     'fields': ['accession', 'compounds', 'type', 'locus'],
     'dir_name': 'mibig'},
    'hits': {},
    'aliases': {}},
   {'database': {'name': 'antiSMASH-DB',
     'version': '4.0',
     'url': 'https://antismash-db.secondarymetabolites.org/area.html?record=@accession@&start=@start@&end=@end@',
     'id_format': '@accession@',
     'description_format': '@type@: @locus@',
     'fields': ['accession', 'type', 'locus', 'start', 'end'],
     'dir_name': 'asdb'},
    'hits': {},
    'aliases': {}}],
  'aliases': {}}}

In [12]:
data["records"][81]["modules"]["antismash.detection.hmm_detection"]

{'record_id': 'NODE_110_length_7016_cov_6.405832',
 'schema_version': 2,
 'enabled_types': ['phosphoglycolipid',
  'lassopeptide',
  'acyl_amino_acids',
  'hglE-KS',
  'lipolanthine',
  'microviridin',
  'aminopolycarboxylic-acid',
  'nucleoside',
  'NRPS-like',
  'glycocin',
  'T3PKS',
  'transAT-PKS-like',
  'lanthipeptide-class-i',
  'NRP-metallophore',
  'butyrolactone',
  'betalactone',
  'epipeptide',
  'phosphonate',
  'PpyS-KS',
  'PUFA',
  'archaeal-RiPP',
  'tropodithietic-acid',
  'cyanobactin',
  'lanthipeptide-class-v',
  'ranthipeptide',
  'transAT-PKS',
  'mycosporine-like',
  'oligosaccharide',
  'furan',
  'opine-like-metallophore',
  'spliceotide',
  'amglyccycl',
  'fungal-RiPP',
  'pyrrolidine',
  'phosphonate-like',
  'CDPS',
  'isocyanide',
  'ladderane',
  'indole',
  'PBDE',
  'T2PKS',
  'crocagin',
  'prodigiosin',
  'linaridin',
  'ectoine',
  'darobactin',
  'thioamide-NRP',
  'terpene',
  'arylpolyene',
  'bottromycin',
  'hserlactone',
  'thioamitides',
  '

In [13]:
data["records"][81]["modules"]["antismash.detection.cluster_hmmer"]

{'hits': [{'location': '[925:1267](-)',
   'label': 'ctg82_1',
   'locus_tag': 'ctg82_1',
   'domain': 'Response_reg',
   'evalue': 1.5e-09,
   'score': 38.1,
   'identifier': 'PF00072.27',
   'description': 'Response regulator receiver domain',
   'protein_start': 4,
   'protein_end': 118,
   'translation': 'IILCEDDIVLMSHYKSIIDNWIMINDYDMKLVLATTDPEDIFLYLSKNKVTNSLYFLDIDLGKHLDGIDVAQSIRTDNEFAQIVFITSHQELAMETLKRQIAPLDYIVKDEQSE'},
  {'location': '[550:826](-)',
   'label': 'ctg82_1',
   'locus_tag': 'ctg82_1',
   'domain': 'LytTR',
   'evalue': 5.5e-15,
   'score': 55.5,
   'identifier': 'PF04397.18',
   'description': 'LytTr DNA-binding domain',
   'protein_start': 151,
   'protein_end': 243,
   'translation': 'NFRVDITSIYFLETSVTPHKIILYGENMMYEFYGKMNDLEEAYPELLRVHKSFLINLNKVISVDYKSRMIFFPEEYSCKFPLAKKRILRKKL'},
  {'location': '[1295:1607](-)',
   'label': 'ctg82_2',
   'locus_tag': 'ctg82_2',
   'domain': 'HATPase_c_5',
   'evalue': 7.5e-11,
   'score': 41.9,
   'identifier': 'PF14501.9',
   'des

In [14]:
data["records"][81]["modules"]["antismash.modules.clusterblast"]

{'schema_version': 2,
 'record_id': 'NODE_110_length_7016_cov_6.405832',
 'general': {'record_id': 'NODE_110_length_7016_cov_6.405832',
  'schema_version': 3,
  'results': [{'region_number': 1,
    'total_hits': 298,
    'ranking': [[{'accession': 'NZ_KB946108',
       'cluster_label': 'c78814-95380',
       'proteins': ['WP_002310545.1',
        'WP_002295975.1',
        'WP_002287505.1',
        'UA3_RS15980',
        'WP_010733060.1',
        'WP_010733061.1',
        'WP_010733062.1',
        'WP_010733063.1',
        'WP_010733064.1',
        'WP_010733065.1',
        'WP_010733066.1',
        'WP_010733067.1',
        'WP_179189718.1',
        'WP_229216198.1',
        'WP_010733068.1',
        'WP_010733069.1',
        'WP_010733070.1',
        'WP_010733071.1',
        'WP_042957099.1',
        'WP_010733073.1'],
       'description': 'Enterococcus faecium EnGen0263 strain UAA1280 acvJW-supercont1.3, whole genome shotgun sequence',
       'cluster_type': 'RiPP-like',
       'ta

In [15]:
data["records"][81]["modules"]["antismash.modules.clusterblast"]["general"]["results"][0]["ranking"][1][1]["pairings"][0]

['input|c1|520-1279|-|ctg82_1|',
 0,
 {'name': 'A5816_RS11485',
  'genecluster': 'NZ_NGLI01000002_c555810-572363',
  'start': 564799,
  'end': 565557,
  'strand': '+',
  'annotation': 'LytTR_family_DNA-binding_domain-containing_protein',
  'perc_ident': 93,
  'blastscore': 460,
  'perc_coverage': 98.80952380952381,
  'evalue': 1.94e-164,
  'locus_tag': 'WP_010733065.1'}]

In [16]:
data["records"][81]["modules"]["antismash.modules.clusterblast"]["general"]["results"][0]["ranking"][1][1].keys()

dict_keys(['hits', 'core_gene_hits', 'blast_score', 'synteny_score', 'core_bonus', 'pairings', 'similarity'])

In [17]:
data["records"][81]["modules"]["antismash.modules.clusterblast"].keys()

dict_keys(['schema_version', 'record_id', 'general', 'subcluster', 'knowncluster'])

In [18]:
data["records"][81]["modules"]["antismash.modules.clusterblast"]["record_id"]

'NODE_110_length_7016_cov_6.405832'

In [19]:
data["records"][81]["modules"]["antismash.modules.clusterblast"]["knowncluster"].keys()

dict_keys(['record_id', 'schema_version', 'results', 'proteins', 'search_type', 'data_version', 'mibig_entries'])

In [20]:
data["records"][81]["modules"]["antismash.modules.clusterblast"]["knowncluster"]["mibig_entries"]["1"].keys()

dict_keys(['ctg82_1', 'ctg82_2', 'ctg82_3'])

In [21]:
data["records"][81]["modules"]["antismash.modules.clusterblast"].keys()

dict_keys(['schema_version', 'record_id', 'general', 'subcluster', 'knowncluster'])

In [22]:
data["records"][81]["modules"]["antismash.modules.clusterblast"]["general"].keys()

dict_keys(['record_id', 'schema_version', 'results', 'proteins', 'search_type'])

In [23]:
data["records"][81]["modules"]["antismash.modules.clusterblast"]["general"]["results"][0].keys()

dict_keys(['region_number', 'total_hits', 'ranking', 'prefix'])

In [24]:
data["records"][81]["modules"]["antismash.modules.clusterblast"]["general"]["results"][0]

{'region_number': 1,
 'total_hits': 298,
 'ranking': [[{'accession': 'NZ_KB946108',
    'cluster_label': 'c78814-95380',
    'proteins': ['WP_002310545.1',
     'WP_002295975.1',
     'WP_002287505.1',
     'UA3_RS15980',
     'WP_010733060.1',
     'WP_010733061.1',
     'WP_010733062.1',
     'WP_010733063.1',
     'WP_010733064.1',
     'WP_010733065.1',
     'WP_010733066.1',
     'WP_010733067.1',
     'WP_179189718.1',
     'WP_229216198.1',
     'WP_010733068.1',
     'WP_010733069.1',
     'WP_010733070.1',
     'WP_010733071.1',
     'WP_042957099.1',
     'WP_010733073.1'],
    'description': 'Enterococcus faecium EnGen0263 strain UAA1280 acvJW-supercont1.3, whole genome shotgun sequence',
    'cluster_type': 'RiPP-like',
    'tags': ['UA3_RS14245',
     'UA3_RS14250',
     'UA3_RS14255',
     'UA3_RS15980',
     'UA3_RS14260',
     'UA3_RS15985',
     'UA3_RS16685',
     'UA3_RS14265',
     'UA3_RS14270',
     'UA3_RS14275',
     'UA3_RS14280',
     'UA3_RS14285',
     'UA3_

In [25]:
len(data["records"][81]["modules"]["antismash.modules.clusterblast"]["general"]["results"])

1

In [26]:
data["records"][81]["modules"]["antismash.modules.clusterblast"]["general"]["results"]

[{'region_number': 1,
  'total_hits': 298,
  'ranking': [[{'accession': 'NZ_KB946108',
     'cluster_label': 'c78814-95380',
     'proteins': ['WP_002310545.1',
      'WP_002295975.1',
      'WP_002287505.1',
      'UA3_RS15980',
      'WP_010733060.1',
      'WP_010733061.1',
      'WP_010733062.1',
      'WP_010733063.1',
      'WP_010733064.1',
      'WP_010733065.1',
      'WP_010733066.1',
      'WP_010733067.1',
      'WP_179189718.1',
      'WP_229216198.1',
      'WP_010733068.1',
      'WP_010733069.1',
      'WP_010733070.1',
      'WP_010733071.1',
      'WP_042957099.1',
      'WP_010733073.1'],
     'description': 'Enterococcus faecium EnGen0263 strain UAA1280 acvJW-supercont1.3, whole genome shotgun sequence',
     'cluster_type': 'RiPP-like',
     'tags': ['UA3_RS14245',
      'UA3_RS14250',
      'UA3_RS14255',
      'UA3_RS15980',
      'UA3_RS14260',
      'UA3_RS15985',
      'UA3_RS16685',
      'UA3_RS14265',
      'UA3_RS14270',
      'UA3_RS14275',
      'UA3_RS1

In [27]:
data["records"][58]["modules"]["antismash.modules.clusterblast"]["knowncluster"]["mibig_entries"]["1"]

{'ctg59_6': [['AAO65813.1',
   'hypothetical_transport_protein',
   'BGC0000100',
   1,
   'Polyketide',
   32.0,
   156.0,
   91.21621621621621,
   2.23e-45]]}

In [28]:
for ix, vl in data["records"][58]["modules"]["antismash.modules.clusterblast"]["knowncluster"]["mibig_entries"]["1"].items():
    ctg_tagid = ix
    mibig_entries = vl

In [29]:
with open(file_path_1, 'r') as files:
    data = json.load(files)
    
    file_name = data["input_file"]
    index_ = above_two(data["records"])

    for i in index_:
        print(data["records"][i]["modules"]["antismash.modules.clusterblast"]["record_id"])


NODE_83_length_8923_cov_7.096640
NODE_110_length_7016_cov_6.405832


In [30]:
# import csv

# # Provided data structure
# data = {
#     'hits': 3,
#     'core_gene_hits': 1,
#     'blast_score': 1781.0,
#     'synteny_score': 3,
#     'core_bonus': 3,
#     'pairings': [
#         ['input|c1|520-1279|-|ctg82_1|',
#          0,
#          {
#              'name': 'UAU_RS17150',
#              'genecluster': 'NZ_KB946310_c115445-132208',
#              'start': 122189,
#              'end': 122938,
#              'strand': '-',
#              'annotation': 'LytTR_family_DNA-binding_domain-containing_protein',
#              'perc_ident': 75,
#              'blastscore': 392,
#              'perc_coverage': 99.20634920634922,
#              'evalue': 1.14e-137,
#              'locus_tag': 'WP_010758328.1'
#          }]
#     ]
# }

# # Extracting data
# hits = data['hits']
# blast_score = data['blast_score']
# core_bonus = data['core_bonus']
# pairings = data['pairings']

# csv_filename = 'output.csv'
# with open(csv_filename, 'w', newline='') as csvfile:
#     writer = csv.writer(csvfile)
#     writer.writerow(['hits', 'blast_score', 'core_bonus', 'input_id', 'index', 'name', 'genecluster', 'start', 'end', 'strand', 'annotation', 'perc_ident', 'blastscore', 'perc_coverage', 'evalue', 'locus_tag'])
#     for pairing in pairings:
#         input_id, index, details = pairing
#         writer.writerow([hits, blast_score, core_bonus, input_id, index,
#                          details['name'], details['genecluster'], details['start'], details['end'],
#                          details['strand'], details['annotation'], details['perc_ident'],
#                          details['blastscore'], details['perc_coverage'], details['evalue'],
#                          details['locus_tag']])

# print(f"CSV file '{csv_filename}' has been created.")


In [31]:
import csv

with open(file_path_1, 'r') as files:
    data = json.load(files)
    
    file_name = data["input_file"]
    index_ = above_two(data["records"])

    csv_filename = 'output.csv'
    with open(csv_filename, 'w', newline='') as csvfile:
        writer = csv.writer(csvfile)
        writer.writerow(["Bind_ID", "Contig_ID", "Gene_Coordinates", "Ctg_Number", "Ctg_Tag_Id", "Mibig_biosynthetic_gene_cluster", "Product", "Blast_name"])
        for i in index_:
            bind_id = data["input_file"]
            contig_id = data["records"][i]["modules"]["antismash.modules.clusterblast"]["record_id"]
            blast_info = data["records"][i]["modules"]["antismash.modules.clusterblast"]["general"]["results"][0]["ranking"][1][1]["pairings"]

            for j in range(len(blast_info)):
                ctg_info = blast_info[j][0]
                blast_name, blast_genecluster, blast_start, blast_end, blast_strand, blast_annotation, blast_perc_ident, blast_blastscore, blast_perc_coverage, blast_evalue, blast_locus_tag  = blast_info[j][2].values()

                ctg_cordinates = ctg_info.split("|")[2]
                ctg_name = ctg_info.split("|")[4]

                for ix, vl in data["records"][i]["modules"]["antismash.modules.clusterblast"]["knowncluster"]["mibig_entries"]["1"].items():
                    ctg_tagid = ix
                    ctg_id = ctg_tagid[:5]

                    # Only first result
                    mibig_entries = vl[0]

                    # mibig info
                    biosynthetic_gene_cluster_id = mibig_entries[2]
                    mibig_entries_product = mibig_entries[4]


                    if (ctg_name == ctg_tagid):
                        writer.writerow([bind_id, contig_id,"", ctg_id, ctg_tagid, biosynthetic_gene_cluster_id, mibig_entries_product, blast_name])
                    else:
                        writer.writerow([bind_id, contig_id,"", ctg_id, ctg_tagid, biosynthetic_gene_cluster_id, mibig_entries_product, ""])
                    print(mibig_entries)
            


['AAO65813.1', 'hypothetical_transport_protein', 'BGC0000100', 1, 'Polyketide', 32.0, 156.0, 91.21621621621621, 2.23e-45]
['AAO65813.1', 'hypothetical_transport_protein', 'BGC0000100', 1, 'Polyketide', 32.0, 156.0, 91.21621621621621, 2.23e-45]
['MBX4265659.1', 'LytTR_family_DNA-binding_domain-containing_protein', 'BGC0002667', 1, 'RiPP', 34.0, 127.0, 90.47619047619048, 2.17e-35]
['ABQ84446.1', 'sensory_transduction_histidine_kinase', 'BGC0000624', 1, 'RiPP', 35.0, 115.0, 43.7070938215103, 6.31e-28]
['AAB81307.1', 'ATP-dependent_translocator', 'BGC0002579', 1, 'RiPP', 52.0, 732.0, 99.44674965421854, 9.1e-259]
['MBX4265659.1', 'LytTR_family_DNA-binding_domain-containing_protein', 'BGC0002667', 1, 'RiPP', 34.0, 127.0, 90.47619047619048, 2.17e-35]
['ABQ84446.1', 'sensory_transduction_histidine_kinase', 'BGC0000624', 1, 'RiPP', 35.0, 115.0, 43.7070938215103, 6.31e-28]
['AAB81307.1', 'ATP-dependent_translocator', 'BGC0002579', 1, 'RiPP', 52.0, 732.0, 99.44674965421854, 9.1e-259]
['MBX4265659

In [32]:
import pandas as pd
pd.read_csv("./output.csv")

Bind_ID                          Contig_ID  Gene_Coordinates  \
0   MG_323757.bin1.fa   NODE_83_length_8923_cov_7.096640               NaN   
1   MG_323757.bin1.fa   NODE_83_length_8923_cov_7.096640               NaN   
2   MG_323757.bin1.fa  NODE_110_length_7016_cov_6.405832               NaN   
3   MG_323757.bin1.fa  NODE_110_length_7016_cov_6.405832               NaN   
4   MG_323757.bin1.fa  NODE_110_length_7016_cov_6.405832               NaN   
5   MG_323757.bin1.fa  NODE_110_length_7016_cov_6.405832               NaN   
6   MG_323757.bin1.fa  NODE_110_length_7016_cov_6.405832               NaN   
7   MG_323757.bin1.fa  NODE_110_length_7016_cov_6.405832               NaN   
8   MG_323757.bin1.fa  NODE_110_length_7016_cov_6.405832               NaN   
9   MG_323757.bin1.fa  NODE_110_length_7016_cov_6.405832               NaN   
10  MG_323757.bin1.fa  NODE_110_length_7016_cov_6.405832               NaN   

   Ctg_Number Ctg_Tag_Id Mibig_biosynthetic_gene_cluster     Product  \
0       ctg59    ctg59_6                      BGC0000100  Polyketide   
1       ctg59    ctg59_6                      BGC0000100  Polyketide   
2       ctg82    ctg82_1                      BGC0002667        RiPP   
3       ctg82    ctg82_2                      BGC0000624        RiPP   
4       ctg82    ctg82_3                      BGC0002579        RiPP   
5       ctg82    ctg82_1                      BGC0002667        RiPP   
6       ctg82    ctg82_2                      BGC0000624        RiPP   
7       ctg82    ctg82_3                      BGC0002579        RiPP   
8       ctg82    ctg82_1                      BGC0002667        RiPP   
9       ctg82    ctg82_2                      BGC0000624        RiPP   
10      ctg82    ctg82_3                      BGC0002579        RiPP   

       Blast_name  
0             NaN  
1             NaN  
2   A5816_RS11485  
3             NaN  
4             NaN  
5             NaN  
6   A5816_RS11480  
7             NaN  
8             NaN  
9             NaN  
10  A5816_RS11475

In [33]:
data["records"][81]["modules"]["antismash.modules.clusterblast"]["knowncluster"]["mibig_entries"]["1"].keys()


dict_keys(['ctg82_1', 'ctg82_2', 'ctg82_3'])

# MIBIG ENTRIES

1. Accession Number - AAO65813.1
2. Identifier for the biosynthetic gene cluster (BGC) - BGC0000100
3. Position of this particular gene or protein - 1
4. Biosynthetic pathway associated with this cluster - Polyketide
5. Length of the protein or gene sequence - 32
6. Location or position of the gene or protein on the contig - 156 
7. Percent identity or similarity of the identified protein - 91.21
8. E-value, which is a measure of the statistical significance of the match - 2.23e-45


# Cluster Blast Json Extractions

In [34]:
blast_info = data["records"][58]["modules"]["antismash.modules.clusterblast"]["general"]["results"][0]["ranking"][1][1]["pairings"]

for i in range(len(blast_info)):
    ctg_info = blast_info[i][0]
    blast_name, blast_genecluster, blast_start, blast_end, blast_strand, blast_annotation, blast_perc_ident, blast_blastscore, blast_perc_coverage, blast_evalue, blast_locus_tag  = blast_info[i][2].values()

    ctg_cordinates = ctg_info.split("|")[2]
    ctg_name = ctg_info.split("|")[4]
    
    print(ctg_name)

ctg59_1
ctg59_2


In [35]:
blast_name

'KW861_RS14645'

In [36]:
blast_info[0]

['input|c1|603-801|+|ctg59_1|',
 0,
 {'name': 'KW861_RS14650',
  'genecluster': 'NZ_CP078505_c3139533-3154876',
  'start': 3144729,
  'end': 3144962,
  'strand': '-',
  'annotation': 'Blp_family_class_II_bacteriocin',
  'perc_ident': 92,
  'blastscore': 49,
  'perc_coverage': 95.38461538461539,
  'evalue': 1.64e-07,
  'locus_tag': 'WP_162781304.1'}]

In [37]:
ctg_info = data["records"][81]["modules"]["antismash.modules.clusterblast"]["general"]["results"][0]["ranking"][1][1]["pairings"][2][0]

In [38]:
ctg_cordinates = ctg_info.split("|")[2]
ctg_name = ctg_info.split("|")[4]

ctg_name

'ctg82_3'

In [39]:
blast_info = data["records"][81]["modules"]["antismash.modules.clusterblast"]["general"]["results"][0]["ranking"][1][1]["pairings"][0][2]

In [40]:
blast_info

{'name': 'A5816_RS11485',
 'genecluster': 'NZ_NGLI01000002_c555810-572363',
 'start': 564799,
 'end': 565557,
 'strand': '+',
 'annotation': 'LytTR_family_DNA-binding_domain-containing_protein',
 'perc_ident': 93,
 'blastscore': 460,
 'perc_coverage': 98.80952380952381,
 'evalue': 1.94e-164,
 'locus_tag': 'WP_010733065.1'}

In [41]:
blast_name, blast_genecluster, blast_start, blast_end, blast_strand, blast_annotation, blast_perc_ident, blast_blastscore, blast_perc_coverage, blast_evalue, blast_locus_tag  = blast_info.values()

In [42]:
blast_locus_tag

'WP_010733065.1'

In [43]:
name, genecluster, start, end, strand, annotation, perc_ident, blastscore, perc_coverage, evalue, locus_tag = data["records"][81]["modules"]["antismash.modules.clusterblast"]["general"]["results"][0]["ranking"][1][1]["pairings"][0][2]

In [44]:
name

'name'

In [45]:
genecluster

'genecluster'

In [46]:
start

'start'

In [97]:
import json
import csv

def above_two(data_records):
    index=[]
    for i in range(len(data_records)):
        modules = data_records[i]["modules"].keys()
        if len(modules) > 2:
            # print(i, len(modules))
            index.append(i)
        else:
            continue
    return index

def process_data(file_path_1):
    with open(file_path_1, 'r') as files:
        data = json.load(files)
        index_ = above_two(data["records"])

        csv_filename = 'output.csv'
        with open(csv_filename, 'w', newline='') as csvfile:
            writer = csv.writer(csvfile)
            writer.writerow(["Bin_ID", "Contig_ID", "Gene_Coordinates", "Ctg_Number", "Ctg_Tag_Id", "Mibig_biosynthetic_gene_cluster", "Mibig_Product", "Blast_name", "Blast_genecluster","Blast_annotation","Blast_perc_coverage", "Blast_perc_ident", "Blast_blastscore", "Blast_evalue", "Blast_locus_tag" ])
            
            for i in index_:
                bin_id = data["input_file"]
                contig_id = data["records"][i]["modules"]["antismash.modules.clusterblast"]["record_id"]
                blast_info = data["records"][i]["modules"]["antismash.modules.clusterblast"]["general"]["results"][0]["ranking"][1][1]["pairings"]
                mibig_entries = data["records"][i]["modules"]["antismash.modules.clusterblast"]["knowncluster"]["mibig_entries"]["1"]

                # Dict for each ctg_tagid
                blast_ = {}
                for blast in blast_info:
                    ctg_name = blast[0].split("|")[4]
                    # ctg_cordinates = blast[0].split("|")[2]
                    blast_["ctg_cordinates"] = blast[0].split("|")[2]
                    blast_name, blast_genecluster, blast_start, blast_end, blast_strand, blast_annotation, blast_perc_ident, blast_blastscore, blast_perc_coverage, blast_evalue, blast_locus_tag  = blast[2].values()
                    blast_[ctg_name] = {"blast_name" : blast_name, 
                                        "blast_genecluster": blast_genecluster,
                                        "blast_annotation": blast_annotation,
                                        "blast_perc_coverage":blast_perc_coverage,
                                        "blast_perc_ident": blast_perc_ident,
                                        "blast_blastscore": blast_blastscore,
                                        "blast_evalue": blast_evalue,
                                        "blast_locus_tag": blast_locus_tag,
                                        "cordinates": blast[0].split("|")[2] }

                # print(blast_)

                blast_keys = list(blast_.keys())[1:]
                mibig_entries_keys = [ ctg_tagid for ctg_tagid in mibig_entries.keys()]
                # print(mibig_entries_keys)
                # print("blast keys")
                # print(blast_keys)

                for ix, vl in  enumerate(blast_keys):
                    print(ix)
                    ctg_id = vl.split("_")[0]
                    ctg_tagid = vl
                    ctg_cordinates = blast_[vl]["cordinates"]
                    blast_name = blast_[vl]["blast_name"]
                    blast_genecluster= blast_[vl]["blast_genecluster"]
                    blast_annotation = blast_[vl]["blast_annotation"]
                    blast_perc_coverage = blast_[vl]["blast_perc_coverage"]
                    blast_perc_ident = blast_[vl]["blast_perc_ident"]
                    blast_blastscore = blast_[vl]["blast_blastscore"]
                    blast_evalue = blast_[vl]["blast_evalue"]
                    blast_locus_tag = blast_[vl]["blast_locus_tag"]
                    biosynthetic_gene_cluster_id=None 
                    mibig_entries_product=None

                    if vl not in mibig_entries_keys:
                        writer.writerow([
                                bin_id, 
                                contig_id,
                                ctg_cordinates, 
                                ctg_id, 
                                ctg_tagid, 
                                biosynthetic_gene_cluster_id, 
                                mibig_entries_product, 
                                blast_name,
                                blast_genecluster,
                                blast_annotation,
                                blast_perc_coverage,
                                blast_perc_ident,
                                blast_blastscore,
                                blast_evalue,
                                blast_locus_tag
                            ])



                for ctg_tagid, vl in mibig_entries.items():
                    ctg_id = ctg_tagid.split("_")[0]
                    mibig_entry = vl[0]
                    biosynthetic_gene_cluster_id = mibig_entry[2]
                    mibig_entries_product = mibig_entry[4]
                    blast_info_ = blast_.get(ctg_tagid)

                    if blast_info_:
                        ctg_cordinates = blast_info_["cordinates"]
                        blast_name = blast_info_["blast_name"]
                        blast_genecluster= blast_info_["blast_genecluster"]
                        blast_annotation = blast_info_["blast_annotation"]
                        blast_perc_coverage = blast_info_["blast_perc_coverage"]
                        blast_perc_ident = blast_info_["blast_perc_ident"]
                        blast_blastscore = blast_info_["blast_blastscore"]
                        blast_evalue = blast_info_["blast_evalue"]
                        blast_locus_tag = blast_info_["blast_locus_tag"]

                    else:
                        ctg_cordinates = None
                        blast_name = None
                        blast_genecluster= None
                        blast_annotation = None
                        blast_perc_coverage = None
                        blast_perc_ident = None
                        blast_blastscore = None
                        blast_evalue = None
                        blast_locus_tag = None
                        
                    writer.writerow([
                        bin_id, 
                        contig_id,
                        ctg_cordinates, 
                        ctg_id, 
                        ctg_tagid, 
                        biosynthetic_gene_cluster_id, 
                        mibig_entries_product, 
                        blast_name,
                        blast_genecluster,
                        blast_annotation,
                        blast_perc_coverage,
                        blast_perc_ident,
                        blast_blastscore,
                        blast_evalue,
                        blast_locus_tag
                    ])

process_data(file_path_1)

['ctg59_6']
blast keys
['ctg59_1', 'ctg59_2']
0
1
['ctg82_1', 'ctg82_2', 'ctg82_3']
blast keys
['ctg82_1', 'ctg82_2', 'ctg82_3']
0
1
2


In [98]:
import pandas as pd
pd.read_csv("./output.csv")

Bin_ID                          Contig_ID Gene_Coordinates  \
0  MG_323757.bin1.fa   NODE_83_length_8923_cov_7.096640          603-801   
1  MG_323757.bin1.fa   NODE_83_length_8923_cov_7.096640          813-996   
2  MG_323757.bin1.fa   NODE_83_length_8923_cov_7.096640              NaN   
3  MG_323757.bin1.fa  NODE_110_length_7016_cov_6.405832         520-1279   
4  MG_323757.bin1.fa  NODE_110_length_7016_cov_6.405832        1280-2594   
5  MG_323757.bin1.fa  NODE_110_length_7016_cov_6.405832        2682-4854   

  Ctg_Number Ctg_Tag_Id Mibig_biosynthetic_gene_cluster Mibig_Product  \
0      ctg59    ctg59_1                             NaN           NaN   
1      ctg59    ctg59_2                             NaN           NaN   
2      ctg59    ctg59_6                      BGC0000100    Polyketide   
3      ctg82    ctg82_1                      BGC0002667          RiPP   
4      ctg82    ctg82_2                      BGC0000624          RiPP   
5      ctg82    ctg82_3                      BGC0002579          RiPP   

      Blast_name               Blast_genecluster  \
0  KW861_RS14650    NZ_CP078505_c3139533-3154876   
1  KW861_RS14645    NZ_CP078505_c3139533-3154876   
2            NaN                             NaN   
3  A5816_RS11485  NZ_NGLI01000002_c555810-572363   
4  A5816_RS11480  NZ_NGLI01000002_c555810-572363   
5  A5816_RS11475  NZ_NGLI01000002_c555810-572363   

                                    Blast_annotation  Blast_perc_coverage  \
0                    Blp_family_class_II_bacteriocin            95.384615   
1                    Blp_family_class_II_bacteriocin           100.000000   
2                                                NaN                  NaN   
3  LytTR_family_DNA-binding_domain-containing_pro...            98.809524   
4                     GHKL_domain-containing_protein            72.311213   
5            peptide_cleavage/export_ABC_transporter           100.000000   

   Blast_perc_ident  Blast_blastscore   Blast_evalue Blast_locus_tag  
0              92.0              49.0   1.640000e-07  WP_162781304.1  
1              83.0              80.0   5.870000e-20  WP_103301327.1  
2               NaN               NaN            NaN             NaN  
3              93.0             460.0  1.940000e-164  WP_010733065.1  
4              88.0             527.0  6.040000e-187  WP_080114483.1  
5              91.0            1263.0   0.000000e+00  WP_010733063.1

In [49]:
directory_path = "./results/"

import os 
os.listdir(directory_path)

['MG_323829.bin2',
 'MG_323757.bin2',
 '.DS_Store',
 'MG_323757.bin3',
 'MG_323829.bin3',
 'MG_323829.bin1',
 'MG_323757.bin1']

In [50]:
import json
import csv

def above_two(data_records):
    index=[]
    for i in range(len(data_records)):
        modules = data_records[i]["modules"].keys()
        if len(modules) > 2:
            # print(i, len(modules))
            index.append(i)
        else:
            continue
    return index



def process_data(file_path_1):

    csv_filename = 'combined_output.csv'
    
    # Check if the output file already exists
    file_exists = os.path.isfile(csv_filename)

    
    with open(file_path_1, 'r') as files:
        data = json.load(files)
        index_ = above_two(data["records"])

        csv_filename = 'output.csv'
        with open(csv_filename, 'w', newline='') as csvfile:
            writer = csv.writer(csvfile)
            writer.writerow(["Bin_ID", "Contig_ID", "Gene_Coordinates", "Ctg_Number", "Ctg_Tag_Id", "Mibig_biosynthetic_gene_cluster", "Mibig_Product", "Blast_name", "Blast_genecluster","Blast_annotation","Blast_perc_coverage", "Blast_perc_ident", "Blast_blastscore", "Blast_evalue", "Blast_locus_tag" ])
            
            for i in index_:
                bin_id = data["input_file"]
                contig_id = data["records"][i]["modules"]["antismash.modules.clusterblast"]["record_id"]
                blast_info = data["records"][i]["modules"]["antismash.modules.clusterblast"]["general"]["results"][0]["ranking"][1][1]["pairings"]
                mibig_entries = data["records"][i]["modules"]["antismash.modules.clusterblast"]["knowncluster"]["mibig_entries"]["1"]

                # Dict for each ctg_tagid
                blast_ = {}
                for blast in blast_info:
                    ctg_name = blast[0].split("|")[4]
                    # ctg_cordinates = blast[0].split("|")[2]
                    blast_["ctg_cordinates"] = blast[0].split("|")[2]
                    blast_name, blast_genecluster, blast_start, blast_end, blast_strand, blast_annotation, blast_perc_ident, blast_blastscore, blast_perc_coverage, blast_evalue, blast_locus_tag  = blast[2].values()
                    blast_[ctg_name] = {"blast_name" : blast_name, 
                                        "blast_genecluster": blast_genecluster,
                                        "blast_annotation": blast_annotation,
                                        "blast_perc_coverage":blast_perc_coverage,
                                        "blast_perc_ident": blast_perc_ident,
                                        "blast_blastscore": blast_blastscore,
                                        "blast_evalue": blast_evalue,
                                        "blast_locus_tag": blast_locus_tag,
                                        "cordinates": blast[0].split("|")[2] }

                # print(blast_)

                for ctg_tagid, vl in mibig_entries.items():
                    ctg_id = ctg_tagid.split("_")[0]
                    mibig_entry = vl[0]
                    biosynthetic_gene_cluster_id = mibig_entry[2]
                    mibig_entries_product = mibig_entry[4]
                    blast_info_ = blast_.get(ctg_tagid)

                    if blast_info_:
                        ctg_cordinates = blast_info_["cordinates"]
                        blast_name = blast_info_["blast_name"]
                        blast_genecluster= blast_info_["blast_genecluster"]
                        blast_annotation = blast_info_["blast_annotation"]
                        blast_perc_coverage = blast_info_["blast_perc_coverage"]
                        blast_perc_ident = blast_info_["blast_perc_ident"]
                        blast_blastscore = blast_info_["blast_blastscore"]
                        blast_evalue = blast_info_["blast_evalue"]
                        blast_locus_tag = blast_info_["blast_locus_tag"]

                    else:
                        ctg_cordinates = None
                        blast_name = None
                        blast_genecluster= None
                        blast_annotation = None
                        blast_perc_coverage = None
                        blast_perc_ident = None
                        blast_blastscore = None
                        blast_evalue = None
                        blast_locus_tag = None
                        
                    writer.writerow([
                        bin_id, 
                        contig_id,
                        ctg_cordinates, 
                        ctg_id, 
                        ctg_tagid, 
                        biosynthetic_gene_cluster_id, 
                        mibig_entries_product, 
                        blast_name,
                        blast_genecluster,
                        blast_annotation,
                        blast_perc_coverage,
                        blast_perc_ident,
                        blast_blastscore,
                        blast_evalue,
                        blast_locus_tag
                    ])

process_data(file_path_1)

In [51]:
def list_json_files(directory):
    json_files = []
    for root, dirs, files in os.walk(directory):
        for file in files:
            if file.endswith('.json'):
                json_files.append(os.path.join(root, file))
    return json_files

list_json_files(directory_path)

['./results/MG_323829.bin2/MG_323829.bin2.json',
 './results/MG_323757.bin2/MG_323757.bin2.json',
 './results/MG_323757.bin3/MG_323757.bin3.json',
 './results/MG_323829.bin3/MG_323829.bin3.json',
 './results/MG_323829.bin1/MG_323829.bin1.json',
 './results/MG_323757.bin1/MG_323757.bin1.json']

In [52]:
list_dir = ['./results/MG_323829.bin2/MG_323829.bin2.json',
 './results/MG_323757.bin2/MG_323757.bin2.json',
 './results/MG_323757.bin3/MG_323757.bin3.json',
 './results/MG_323829.bin3/MG_323829.bin3.json',
 './results/MG_323829.bin1/MG_323829.bin1.json',
 "./results/MG_323757.bin1/MG_323757.bin1.json"
 ]

for i in list_dir:
    process_data(i)

In [53]:
import pandas as pd
pd.read_csv("./output.csv")

Bin_ID                          Contig_ID Gene_Coordinates  \
0  MG_323757.bin1.fa   NODE_83_length_8923_cov_7.096640              NaN   
1  MG_323757.bin1.fa  NODE_110_length_7016_cov_6.405832         520-1279   
2  MG_323757.bin1.fa  NODE_110_length_7016_cov_6.405832        1280-2594   
3  MG_323757.bin1.fa  NODE_110_length_7016_cov_6.405832        2682-4854   

  Ctg_Number Ctg_Tag_Id Mibig_biosynthetic_gene_cluster Mibig_Product  \
0      ctg59    ctg59_6                      BGC0000100    Polyketide   
1      ctg82    ctg82_1                      BGC0002667          RiPP   
2      ctg82    ctg82_2                      BGC0000624          RiPP   
3      ctg82    ctg82_3                      BGC0002579          RiPP   

      Blast_name               Blast_genecluster  \
0            NaN                             NaN   
1  A5816_RS11485  NZ_NGLI01000002_c555810-572363   
2  A5816_RS11480  NZ_NGLI01000002_c555810-572363   
3  A5816_RS11475  NZ_NGLI01000002_c555810-572363   

                                    Blast_annotation  Blast_perc_coverage  \
0                                                NaN                  NaN   
1  LytTR_family_DNA-binding_domain-containing_pro...            98.809524   
2                     GHKL_domain-containing_protein            72.311213   
3            peptide_cleavage/export_ABC_transporter           100.000000   

   Blast_perc_ident  Blast_blastscore   Blast_evalue Blast_locus_tag  
0               NaN               NaN            NaN             NaN  
1              93.0             460.0  1.940000e-164  WP_010733065.1  
2              88.0             527.0  6.040000e-187  WP_080114483.1  
3              91.0            1263.0   0.000000e+00  WP_010733063.1

# Multiple Files

In [54]:
import json
import csv
import os

def above_two(data_records):
    index=[]
    for i in range(len(data_records)):
        modules = data_records[i]["modules"].keys()
        if len(modules) > 2:
            # print(i, len(modules))
            index.append(i)
        else:
            continue
    return index

def process_files(file_paths):
    
    csv_filename = 'combined_output.csv'
    
    # Check if the output file already exists
    file_exists = os.path.isfile(csv_filename)
    # print(file_exists)
    
    with open(csv_filename, 'a', newline='') as csvfile:
        writer = csv.writer(csvfile)
        
        if not file_exists:
            writer.writerow(["Bin_ID", "Contig_ID", "Gene_Coordinates", "Ctg_Number", "Ctg_Tag_Id", "Mibig_biosynthetic_gene_cluster", "Mibig_Product", "Blast_name", "Blast_genecluster","Blast_annotation","Blast_perc_coverage", "Blast_perc_ident", "Blast_blastscore", "Blast_evalue", "Blast_locus_tag" ])
        
        for file_path in file_paths:

            with open(file_path, 'r') as files:
                data = json.load(files)
                index_ = above_two(data["records"])

                
                # with open(csv_filename, 'w', newline='') as csvfile:
                #     writer = csv.writer(csvfile)
                #     writer.writerow(["Bin_ID", "Contig_ID", "Gene_Coordinates", "Ctg_Number", "Ctg_Tag_Id", "Mibig_biosynthetic_gene_cluster", "Mibig_Product", "Blast_name", "Blast_genecluster","Blast_annotation","Blast_perc_coverage", "Blast_perc_ident", "Blast_blastscore", "Blast_evalue", "Blast_locus_tag" ])
                    
                for i in index_:
                    bin_id = data["input_file"]
                    contig_id = data["records"][i]["modules"]["antismash.modules.clusterblast"]["record_id"]
                    blast_info = data["records"][i]["modules"]["antismash.modules.clusterblast"]["general"]["results"][0]["ranking"][1][1]["pairings"]
                    mibig_entries = data["records"][i]["modules"]["antismash.modules.clusterblast"]["knowncluster"]["mibig_entries"]["1"]

                    # Dict for each ctg_tagid
                    blast_ = {}
                    for blast in blast_info:
                        ctg_name = blast[0].split("|")[4]
                        # ctg_cordinates = blast[0].split("|")[2]
                        blast_["ctg_cordinates"] = blast[0].split("|")[2]
                        blast_name, blast_genecluster, blast_start, blast_end, blast_strand, blast_annotation, blast_perc_ident, blast_blastscore, blast_perc_coverage, blast_evalue, blast_locus_tag  = blast[2].values()
                        blast_[ctg_name] = {"blast_name" : blast_name, 
                                            "blast_genecluster": blast_genecluster,
                                            "blast_annotation": blast_annotation,
                                            "blast_perc_coverage":blast_perc_coverage,
                                            "blast_perc_ident": blast_perc_ident,
                                            "blast_blastscore": blast_blastscore,
                                            "blast_evalue": blast_evalue,
                                            "blast_locus_tag": blast_locus_tag,
                                            "cordinates": blast[0].split("|")[2] }

                    # print(blast_)

                    for ctg_tagid, vl in mibig_entries.items():
                        ctg_id = ctg_tagid.split("_")[0]
                        mibig_entry = vl[0]
                        biosynthetic_gene_cluster_id = mibig_entry[2]
                        mibig_entries_product = mibig_entry[4]
                        blast_info_ = blast_.get(ctg_tagid)

                        if blast_info_:
                            ctg_cordinates = blast_info_["cordinates"]
                            blast_name = blast_info_["blast_name"]
                            blast_genecluster= blast_info_["blast_genecluster"]
                            blast_annotation = blast_info_["blast_annotation"]
                            blast_perc_coverage = blast_info_["blast_perc_coverage"]
                            blast_perc_ident = blast_info_["blast_perc_ident"]
                            blast_blastscore = blast_info_["blast_blastscore"]
                            blast_evalue = blast_info_["blast_evalue"]
                            blast_locus_tag = blast_info_["blast_locus_tag"]

                        else:
                            ctg_cordinates = None
                            blast_name = None
                            blast_genecluster= None
                            blast_annotation = None
                            blast_perc_coverage = None
                            blast_perc_ident = None
                            blast_blastscore = None
                            blast_evalue = None
                            blast_locus_tag = None
                            
                        writer.writerow([
                            bin_id, 
                            contig_id,
                            ctg_cordinates, 
                            ctg_id, 
                            ctg_tagid, 
                            biosynthetic_gene_cluster_id, 
                            mibig_entries_product, 
                            blast_name,
                            blast_genecluster,
                            blast_annotation,
                            blast_perc_coverage,
                            blast_perc_ident,
                            blast_blastscore,
                            blast_evalue,
                            blast_locus_tag
                        ])
            

process_files(list_dir)

# Add Blast None Info

In [56]:
import json
import csv
import os

def above_two(data_records):
    index=[]
    for i in range(len(data_records)):
        modules = data_records[i]["modules"].keys()
        if len(modules) > 2:
            # print(i, len(modules))
            index.append(i)
        else:
            continue
    return index

def process_files(file_paths):
    
    csv_filename = 'combined_output.csv'
    
    # Check if the output file already exists
    file_exists = os.path.isfile(csv_filename)
    # print(file_exists)
    
    with open(csv_filename, 'a', newline='') as csvfile:
        writer = csv.writer(csvfile)
        
        if not file_exists:
            writer.writerow(["Bin_ID", "Contig_ID", "Gene_Coordinates", "Ctg_Number", "Ctg_Tag_Id", "Mibig_biosynthetic_gene_cluster", "Mibig_Product", "Blast_name", "Blast_genecluster","Blast_annotation","Blast_perc_coverage", "Blast_perc_ident", "Blast_blastscore", "Blast_evalue", "Blast_locus_tag" ])
        
        for file_path in file_paths:

            with open(file_path, 'r') as files:
                data = json.load(files)
                index_ = above_two(data["records"])

                
                # with open(csv_filename, 'w', newline='') as csvfile:
                #     writer = csv.writer(csvfile)
                #     writer.writerow(["Bin_ID", "Contig_ID", "Gene_Coordinates", "Ctg_Number", "Ctg_Tag_Id", "Mibig_biosynthetic_gene_cluster", "Mibig_Product", "Blast_name", "Blast_genecluster","Blast_annotation","Blast_perc_coverage", "Blast_perc_ident", "Blast_blastscore", "Blast_evalue", "Blast_locus_tag" ])
                    
                for i in index_:
                    bin_id = data["input_file"]
                    contig_id = data["records"][i]["modules"]["antismash.modules.clusterblast"]["record_id"]
                    blast_info = data["records"][i]["modules"]["antismash.modules.clusterblast"]["general"]["results"][0]["ranking"][1][1]["pairings"]
                    mibig_entries = data["records"][i]["modules"]["antismash.modules.clusterblast"]["knowncluster"].get("mibig_entries", {})
                    
                    for blast in blast_info:
                        ctg_tagid = blast[0]
                        ctg_name = ctg_tagid.split("|")[4]
                        ctg_cordinates = blast[0].split("|")[2]
                        blast_name, blast_genecluster, blast_start, blast_end, blast_strand, blast_annotation, blast_perc_ident, blast_blastscore, blast_perc_coverage, blast_evalue, blast_locus_tag  = blast[2].values()
                        
                        # Check if this ctg_tagid is in mibig_entries
                        mibig_entry = mibig_entries.get(ctg_tagid, [None, None, None, None, None])[0]
                        if mibig_entry:
                            biosynthetic_gene_cluster_id = mibig_entry[2]
                            mibig_entries_product = mibig_entry[4]
                        else:
                            biosynthetic_gene_cluster_id = None
                            mibig_entries_product = None
                        
                        writer.writerow([
                            bin_id, 
                            contig_id,
                            ctg_cordinates, 
                            ctg_tagid.split("_")[0], 
                            ctg_tagid, 
                            biosynthetic_gene_cluster_id, 
                            mibig_entries_product, 
                            blast_name,
                            blast_genecluster,
                            blast_annotation,
                            blast_perc_coverage,
                            blast_perc_ident,
                            blast_blastscore,
                            blast_evalue,
                            blast_locus_tag
                        ])
            

process_files(list_dir)

In [57]:
import pandas as pd
pd.read_csv("./combined_output.csv")

Bin_ID                           Contig_ID Gene_Coordinates  \
0    MG_323829.bin2.fa  NODE_3_length_155941_cov_99.724542      28598-29438   
1    MG_323829.bin2.fa  NODE_3_length_155941_cov_99.724542      29585-29864   
2    MG_323829.bin2.fa  NODE_3_length_155941_cov_99.724542      29958-30246   
3    MG_323829.bin2.fa  NODE_3_length_155941_cov_99.724542      30616-31054   
4    MG_323829.bin2.fa  NODE_3_length_155941_cov_99.724542      31058-32072   
..                 ...                                 ...              ...   
97   MG_323757.bin1.fa    NODE_83_length_8923_cov_7.096640          603-801   
98   MG_323757.bin1.fa    NODE_83_length_8923_cov_7.096640          813-996   
99   MG_323757.bin1.fa   NODE_110_length_7016_cov_6.405832         520-1279   
100  MG_323757.bin1.fa   NODE_110_length_7016_cov_6.405832        1280-2594   
101  MG_323757.bin1.fa   NODE_110_length_7016_cov_6.405832        2682-4854   

                      Ctg_Number                       Ctg_Tag_Id  \
0    input|c1|28598-29438|+|ctg3  input|c1|28598-29438|+|ctg3_28|   
1    input|c1|29585-29864|-|ctg3  input|c1|29585-29864|-|ctg3_29|   
2    input|c1|29958-30246|-|ctg3  input|c1|29958-30246|-|ctg3_30|   
3    input|c1|30616-31054|+|ctg3  input|c1|30616-31054|+|ctg3_31|   
4    input|c1|31058-32072|+|ctg3  input|c1|31058-32072|+|ctg3_32|   
..                           ...                              ...   
97      input|c1|603-801|+|ctg59      input|c1|603-801|+|ctg59_1|   
98      input|c1|813-996|+|ctg59      input|c1|813-996|+|ctg59_2|   
99     input|c1|520-1279|-|ctg82     input|c1|520-1279|-|ctg82_1|   
100   input|c1|1280-2594|-|ctg82    input|c1|1280-2594|-|ctg82_2|   
101   input|c1|2682-4854|-|ctg82    input|c1|2682-4854|-|ctg82_3|   

     Mibig_biosynthetic_gene_cluster  Mibig_Product     Blast_name  \
0                                NaN            NaN  CG392_RS04930   
1                                NaN            NaN  CG392_RS04935   
2                                NaN            NaN  CG392_RS04940   
3                                NaN            NaN  CG392_RS04945   
4                                NaN            NaN  CG392_RS04950   
..                               ...            ...            ...   
97                               NaN            NaN  KW861_RS14650   
98                               NaN            NaN  KW861_RS14645   
99                               NaN            NaN  A5816_RS11485   
100                              NaN            NaN  A5816_RS11480   
101                              NaN            NaN  A5816_RS11475   

                  Blast_genecluster  \
0     NZ_NNRS01000004_c78906-120244   
1     NZ_NNRS01000004_c78906-120244   
2     NZ_NNRS01000004_c78906-120244   
3     NZ_NNRS01000004_c78906-120244   
4     NZ_NNRS01000004_c78906-120244   
..                              ...   
97     NZ_CP078505_c3139533-3154876   
98     NZ_CP078505_c3139533-3154876   
99   NZ_NGLI01000002_c555810-572363   
100  NZ_NGLI01000002_c555810-572363   
101  NZ_NGLI01000002_c555810-572363   

                                      Blast_annotation  Blast_perc_coverage  \
0                                    AAA_family_ATPase            99.641577   
1                                 hypothetical_protein            98.913043   
2    type_II_toxin-antitoxin_system_RelB/DinJ_famil...           100.000000   
3                      ORF6N_domain-containing_protein            79.310345   
4                                   Abi_family_protein           100.000000   
..                                                 ...                  ...   
97                     Blp_family_class_II_bacteriocin            95.384615   
98                     Blp_family_class_II_bacteriocin           100.000000   
99   LytTR_family_DNA-binding_domain-containing_pro...            98.809524   
100                     GHKL_domain-containing_protein            72.311213   
101            peptide_cleavage/export_ABC_transporter     